pkg/bench/tpcc/tpcc_bench_test.go

```diff
@@ -25,42 +20,55 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/logstore"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/pgurlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	_ "github.com/cockroachdb/cockroach/pkg/workload/tpcc"
 )
 
-// BenchmarkTPCC runs a benchmark of different mixes of TPCC queries against
-// a single warehouse. It runs the client side of the workload in a subprocess
-// to filter out the client overhead.
+const storePath = "store"
+
+var rewrite = flag.Bool("rewrite", false,
+	"regenerate store files containing the TPC-C schema and data",
+)
+
+// BenchmarkTPCC runs TPC-C transactions against a single warehouse. It runs the
+// client side of the workload in a subprocess so that the client overhead is
+// not included in CPU and heap profiles.
 //
-// Without any special flags, the benchmark will load the data for the single
-// warehouse once in a temp dir and will then copy the store directory for
-// that warehouse to an in-memory store for each sub-benchmark. The --store-dir
-// flag can be used to specify the path to such a store dir to enable faster
-// iteration re-running the test. The --generate-store-dir flag, when combined
-// with --store-dir will generate a new store directory at the specified path.
-// The combination of the two flags is how you bootstrap such a path initially.
+// The benchmark will load the data for the single warehouse from the
+// testdata/store directory. This directory needs to be populated by running the
+// benchmark once with the --rewrite flag. When running each sub-benchmark, the
+// store directory will be copied to an in-memory store. This enabled faster
+// iteration when re-running the benchmark.
 //
-// For example, generate the store directory with:
+// For example, first generate the store directory with:
 //
-//	./dev bench pkg/bench/tpcc -f BenchmarkTPCC --test-args '--generate-store-dir --store-dir=/tmp/benchtpcc'
+//	./dev bench pkg/bench/tpcc -f 'BenchmarkTPCC/literal/mix=newOrder=1$' --rewrite
 //
-// Reuse the store directory with:
+// Future benchmark runs will reuse the store directory:
 //
-//	./dev bench pkg/bench/tpcc -f BenchmarkTPCC --test-args '--store-dir=/tmp/benchtpcc'
+//	./dev bench pkg/bench/tpcc -f BenchmarkTPCC
 //
-// TODO(ajwerner): Consider moving this all to CCL to leverage the import data
-// loader.
+// To regenerate the store direct, simply run the benchmark with the --rewrite
+// flag again.
 func BenchmarkTPCC(b *testing.B) {
 	defer leaktest.AfterTest(b)()
 	defer log.Scope(b).Close(b)
 
-	// Generate TPCC data if necessary.
-	_, storeDir, cleanup := maybeGenerateStoreDir(b)
-	defer cleanup()
+	// Generate TPCC data if requested.
+	if *rewrite {
+		storeDir := datapathutils.TestDataPath(b, storePath)
```

Comment:
a bit of a random place to leave this comment, but currently (I think) this does single-warehouse. Is that sensible? Wouldn't there be a ton of contention? In particular, I think newOrder requires a sequential ID allocation per warehouse, which would become a bottleneck. I'm not sure this is primarily what we want to measure?

(Just want to start the discussion, not necessarily have it here).

Reply:
The client queries are run in a single thread, so there should be no concurrent queries. And that's by design for this micro-benchmark. I want it to show the cost of a query in a more-or-less ideal case, e.g., without contention.